cumulusci/tasks/salesforce/content_documents.py

```diff
@@ -0,0 +1,209 @@
+import base64
+from pathlib import Path
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.utils import process_list_arg
+from cumulusci.core.exceptions import CumulusCIException
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+
+def to_cumulusci_exception(e: SalesforceMalformedRequest) -> CumulusCIException:
+    return CumulusCIException(
+        "; ".join([error.get("message", "Unknown.") for error in e.content])
+    )
+
+
+class InsertContentDocument(BaseSalesforceApiTask):
+    task_docs = """
+Uploads a profile photo for a specified or default User.
+
+Examples
+--------
+
+Upload a profile photo for the default user.
+
+.. code-block:: yaml
+
+    tasks:
+        upload_profile_photo_default:
+            group: Internal storytelling data
+            class_path: cumulusci.tasks.salesforce.users.UploadProfilePhoto
+            description: Uploads a profile photo for the default user.
+            options:
+                photo: storytelling/photos/default.png
+
+Upload a profile photo for a user whose Alias equals ``grace`` or ``walker``, is active, and created today.
+
+.. code-block:: yaml
+
+    tasks:
+        upload_profile_photo_grace:
+            group: Internal storytelling data
+            class_path: cumulusci.tasks.salesforce.users.UploadProfilePhoto
+            description: Uploads a profile photo for Grace.
+            options:
+                photo: storytelling/photos/grace.png
+                where: (Alias = 'grace' OR Alias = 'walker') AND IsActive = true AND CreatedDate = TODAY
+    """
+
+    task_options = {
+        "path": {
+            "description": "Path to the file to upload as a Content Document.",
+            "required": True,
+        },
+        "queries": {
```

Comment:
We should probably (auto) namespace inject the queries

Reply:
@davisagli , any suggestions on how to get this task to have the namespace auto injected?   Are there magic option names to use to plug into the framework?

Reply:
@spelak-salesforce The anonymous Apex task serves as an example:
1. The task has options named `managed` and `namespaced` (the latter should be called `namespaced_org` for best consistency with most tasks; it's called `namespaced` for the anonymous Apex task for backwards compatibility)
2. Those options are processed here: https://github.com/SFDO-Tooling/CumulusCI/blob/main/cumulusci/tasks/apex/anon.py#L99-L110 -- If the options haven't been specified, determine the defaults using org_config.installed_packages and org_config.namespace
3. Call the inject_namespace utility with these options to replace the standard tokens in whatever string needs it: https://github.com/SFDO-Tooling/CumulusCI/blob/main/cumulusci/tasks/apex/anon.py#L112-L118

Reply:
@davisagli  The following is totally unrelated to changes in this pull request and only for discussion humbly sharing an opinion.
> Since namespace injection is a very common feature in cci tasks + (it seems that) generally tasks only want to namespace inject using the "defaults", could `OrgConfig` support a `_inject_namespace` method that uses the defaults to call the `_inject_namespace` function?
> - It'd be super easy for tasks to use not having to do a manual configuration + manual call.
> - This shouldn't have any backwards compatibility issues since this doesn't alter task (instance) methods.
> - New tasks like `InsertContentDocument` can opt to inject namespaces only using this default config values.   We can wait for feedback that users really want to have options to override the namespace injection behavior to do the extra work of a manual calculation + call.

Reply:
@spelak-salesforce I like the idea of an `_inject_namespace` method to make this more reusable. There's already one on the base Metadata ETL task, actually, but putting it on OrgConfig makes sense, at least at first glance. I think the project_config can be accessed by way of the OrgConfig's `keychain`.

Context here is that I realized while working on automatic namespace injection that this logic was a bit repetitive and would be nice to centralize, but I had trouble figuring out a way to do that that would actually make things simpler, especially while still supporting the explicit options and 100% backwards compatibility. But I think your idea has potential, especially for new tasks.

FWIW I _mostly_ agree that the explicit options are mostly for backwards compatibility and don't need to be added for new tasks. One possible reason to continue providing them is that in the context of a metadeploy plan where it's known that the package will always be managed, it might be nice to avoid making unnecessary API calls to the org to check the installed packages.

Reply:
For an alternative, while writing the above, my initial was:
- The base task or base salesforce task always includes "base" task options of the options needed to inject namespaces: `managed`, `namespaced_org`, and `namespace_[to_]inject`.
    - This would change how task options are compiled.   Task implementations can override the base task options, but at least the options would still exist.  Something like:
       ```python
        task_options = self.base_task_options
        task_options.update(self.task_options)
        ```
- The base task then has an instance method called `_inject_namespace` that does all the calculation.
    - At first, I was worried about breaking existing implementations, but then I remember it's python: any implementation using the same method name will just override the base method.
    - Some extra care may need to be taking since we are "overriding" task options.  I think the trick is never to have the base task reset base task options.  `_inject_namespace` always reads the task options for its calculations.  
    -  If the base task options are overridden in a way that breaks `_inject_namespace`, oops!  Don't use `_inject_namespace`.
- Tasks like `Deploy` that use non-standard task options for namespace injection can do the conversion in `_init_options`.

Reply:
@spelak-salesforce It's already pretty easy to copy particular options from another source (whether it's a base task's options, or a standalone dict), so I don't see a compelling reason to add `base_task_options` as a formal part of the framework. This could be something like this:

    namespace_options = {
        "namespace": { ... },
        "managed": { ... },
        "namespaced_org": { ... },
    }

    class MyTask(BaseTask):
        task_options = {
            **namespace_options,
            "another_option": { ... },
        }

Using ** in this way also makes sure that we're constructing a new dict and not mutating the one we're copying options from.

Reply:
Could we store the `namespace_options` + the following `BaseTask` mixin?

```python
from cumulusci.utils import inject_namespace

namespace_options = {
    "managed": {
        "description": "If False, changes namespace_inject to replace tokens with a blank string",
        "required": False,
    },
    "namespaced_org": {
        "description": "If True, the tokens %%%NAMESPACED_ORG%%% and ___NAMESPACED_ORG___ will get replaced with the namespace.  The default is false causing those tokens to get stripped and replaced with an empty string.  Set this if deploying to a namespaced scratch org or packaging org.",
        "required": False,
    },
    "namespace_inject": {
        "description": "If set, the namespace tokens in files and filenames are replaced with the namespace's prefix",
        "required": False,
    },
}

class NamespaceInjectionMixin:

    def inject_namespace(self, name, content):
        namespace = (
            self.options.get("namespace_inject")
            or self.project_config.project__package__namespace
        )
        if "managed" in self.options:
            managed = process_bool_arg(self.options["managed"])
        else:
            managed = (
                bool(namespace) and namespace in self.org_config.installed_packages
            )
        if "namespaced_org" in self.options:
            namespaced_org = process_bool_arg(self.options["namespaced_org"])
        else:
            namespaced_org = bool(namespace) and namespace == self.org_config.namespace

        return inject_namespace(
            name_content
            namespace=namespace,
            managed=managed,
            namespaced_org=namespaced_org,
        )
```

Reply:
@davisagli , I added `cumulusci.tasks.salesforce.namespaces` containing `NamespaceInjectionMixin` + `namespace_injection_options`.   Thoughts?  Thank you.

Reply:
`NamespaceInjectionMixin` has been moved to #2313